src/components/HomeSearchBox/index.tsx

```diff
@@ -1,42 +1,110 @@
+    //TODO: This below data should be obtained from database
+    const suggestions: string[] = ['Alabama', 'Alaska', 'American Samoa', 'Bubble Holmes', 'Contra Hits', 'Amazon']
+
+    const {
+      currentTarget: { value }
+    } = event
+    const newFilteredSuggestionSuggestions = suggestions.filter(
+      (suggestion: string) => suggestion.toLowerCase().indexOf(value.toLowerCase()) > -1
+    )
+
+    setActive(0)
+    setFilteredSuggestion(newFilteredSuggestionSuggestions)
+    setSuggestionShown(true)
+    setSearchValue(value)
+  }
+
+  const handleListItemClick = (event: React.MouseEvent<HTMLLIElement>) => {
+    setActive(0)
+    setFilteredSuggestion([])
+    setSuggestionShown(false)
+    setSearchValue(event.currentTarget.innerText)
+    handleSearchClick()
   }
 
   const handleSearchClick = () => {
     if (searchValue) {
-      router.push('products/${searchValue}')
+      router.push(`products/${searchValue}`)
       setSearchValue('')
     }
   }
 
   const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
-    if (event.key === 'Enter') {
+    if (searchValue && event.key === 'Enter') {
+      setActive(0)
+      setSuggestionShown(false)
+      setSearchValue(filteredSuggestion[active])
       handleSearchClick()
     }
   }
 
+  const renderSearchSuggestion = () => {
+    if (isSuggestionShown && searchValue) {
+      const filterResults = filteredSuggestion.map((suggestion, index) => {
+        let className = index === active ? 'active' : ''
+
+        return (
+          <li className={className} key={suggestion} onClick={handleListItemClick}>
+            {suggestion}
+          </li>
+        )
+      })
+
+      const hasFilteredSuggestionSuggestion: boolean = filteredSuggestion.length > 0
+      const autoComplete = hasFilteredSuggestionSuggestion ? (
+        <ul className="home-search-suggestion__autocomplete">{filterResults}</ul>
+      ) : (
+        <div className="home-search-suggestion__not-found">
+          <em>Not found</em>
+        </div>
+      )
+
+      return <div className="home-search-suggestion">{autoComplete}</div>
+    } else {
+      return <></>
+    }
+  }
+
   return (
-    <div className="home-search-box">
-      <input
-        value={searchValue}
-        type="search"
-        onChange={handleInputChange}
-        onKeyUp={handleKeyPress}
-        placeholder={t('searchPlaceholder')}
-        name="search"
-        className="home-search-box__input"
-      />
-      <button onClick={handleSearchClick} className="home-search-box__button" />
-    </div>
+    <>
+      <div className="home-search-box">
+        <input
+          className="home-search-box__input"
+          name="search"
+          value={searchValue}
+          placeholder={t('searchPlaceholder')}
+          onChange={handleInputChange}
+          onKeyPress={handleKeyPress}
+        />
+        <button onClick={handleSearchClick} className="home-search-box__button" />
```

Comment:
Normally, we align it this way

```suggestion
        <button className="home-search-box__button" onClick={handleSearchClick} />
```